src/boundaries.rs

```diff
@@ -16,204 +30,117 @@ pub mod boundary_type {
-        } // (├, ├) => Some(├)
-        (boundary_type::VERTICAL_RIGHT, boundary_type::HORIZONTAL_DOWN) => {
-            Some(boundary_type::CROSS)
-        } // (├, ┬) => Some(┼)
-        (boundary_type::VERTICAL_RIGHT, boundary_type::HORIZONTAL_UP) => Some(boundary_type::CROSS), // (├, ┴) => Some(┼)
-        (boundary_type::VERTICAL_RIGHT, boundary_type::CROSS) => Some(boundary_type::CROSS), // (├, ┼) => Some(┼)
+        (BOTTOM_LEFT, BOTTOM_LEFT) => Some(BOTTOM_LEFT), // (└, └) => Some(└)
+        (BOTTOM_LEFT, VERTICAL_LEFT) => Some(CROSS),     // (└, ┤) => Some(┼)
+        (BOTTOM_LEFT, VERTICAL_RIGHT) => Some(VERTICAL_RIGHT), // (└, ├) => Some(├)
+        (BOTTOM_LEFT, HORIZONTAL_DOWN) => Some(CROSS),   // (└, ┬) => Some(┼)
+        (BOTTOM_LEFT, HORIZONTAL_UP) => Some(HORIZONTAL_UP), // (└, ┴) => Some(┴)
 
-        (boundary_type::HORIZONTAL_DOWN, boundary_type::HORIZONTAL_DOWN) => {
-            Some(boundary_type::HORIZONTAL_DOWN)
-        } // (┬, ┬) => Some(┬)
-        (boundary_type::HORIZONTAL_DOWN, boundary_type::HORIZONTAL_UP) => {
-            Some(boundary_type::CROSS)
-        } // (┬, ┴) => Some(┼)
-        (boundary_type::HORIZONTAL_DOWN, boundary_type::CROSS) => Some(boundary_type::CROSS), // (┬, ┼) => Some(┼)
+        (VERTICAL_LEFT, VERTICAL_LEFT) => Some(VERTICAL_LEFT), // (┤, ┤) => Some(┤)
+        (VERTICAL_LEFT, VERTICAL_RIGHT) => Some(CROSS),        // (┤, ├) => Some(┼)
+        (VERTICAL_LEFT, HORIZONTAL_DOWN) => Some(CROSS),       // (┤, ┬) => Some(┼)
+        (VERTICAL_LEFT, HORIZONTAL_UP) => Some(HORIZONTAL_UP), // (┤, ┴) => Some(┼)
 
-        (boundary_type::HORIZONTAL_UP, boundary_type::HORIZONTAL_UP) => {
-            Some(boundary_type::HORIZONTAL_UP)
-        } // (┴, ┴) => Some(┴)
-        (boundary_type::HORIZONTAL_UP, boundary_type::CROSS) => Some(boundary_type::CROSS), // (┴, ┼) => Some(┼)
+        (VERTICAL_RIGHT, VERTICAL_RIGHT) => Some(VERTICAL_RIGHT), // (├, ├) => Some(├)
+        (VERTICAL_RIGHT, HORIZONTAL_DOWN) => Some(CROSS),         // (├, ┬) => Some(┼)
+        (VERTICAL_RIGHT, HORIZONTAL_UP) => Some(CROSS),           // (├, ┴) => Some(┼)
 
-        (boundary_type::CROSS, boundary_type::CROSS) => Some(boundary_type::CROSS), // (┼, ┼) => Some(┼)
+        (HORIZONTAL_DOWN, HORIZONTAL_DOWN) => Some(HORIZONTAL_DOWN), // (┬, ┬) => Some(┬)
+        (HORIZONTAL_DOWN, HORIZONTAL_UP) => Some(CROSS),             // (┬, ┴) => Some(┼)
 
-        (_, _) => None,
-    }
-}
+        (HORIZONTAL_UP, HORIZONTAL_UP) => Some(HORIZONTAL_UP), // (┴, ┴) => Some(┴)
 
-fn find_next_symbol(first_symbol: &str, second_symbol: &str) -> Option<&'static str> {
-    if let Some(symbol) = combine_symbols(first_symbol, second_symbol) {
-        Some(symbol)
-    } else {
-        combine_symbols(second_symbol, first_symbol)
+        (a, b) => combine_symbols(b, a), // If we didn't find a match, swap them
     }
 }
 
-#[derive(PartialEq, Eq, Hash, Debug)]
-pub struct Coordinates {
+pub struct ScreenCanvas {
```

Comment:
The thinking here was potentially to have a more generic type for "Rectangle that can contain other rectangles and knows some stuff about what needs to be drawn". This may already be covered by terminal_pane or similar, I'm not sure